components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90

```diff
@@ -728,123 +507,53 @@ subroutine gas_phase_chemdr(lchnk, ncol, imozart, q, &
-
-       call lin_strat_chem_solve( ncol, lchnk, col_dens(:,:,1), tfld, zen_angle, pmid, delt, rlats, troplev, & !in
-            linoz_o3_clim, linoz_t_clim, linoz_o3col_clim, linoz_PmL_clim, linoz_dPmL_dO3, linoz_dPmL_dT, & !in
-            linoz_dPmL_dO3col, linoz_cariolle_psc, & !in
-            vmr(:,:,o3_ndx) ) !in-out
-            
-       call   lin_strat_sfcsink (ncol, lchnk, delt, pdel(:ncol,:), & !in
-            vmr(:,:,o3_ndx)) !in-outs
-    end if
+    call lin_strat_chem_solve( ncol, lchnk, col_dens(:,:,1), tfld, zen_angle, pmid, delt, rlats, troplev, & !in
+       linoz_o3_clim, linoz_t_clim, linoz_o3col_clim, linoz_PmL_clim, linoz_dPmL_dO3, linoz_dPmL_dT, & !in
+       linoz_dPmL_dO3col, linoz_cariolle_psc, & !in
+       vmr(:,:,o3_ndx) ) !in-out
+      
+    call   lin_strat_sfcsink (ncol, lchnk, delt, pdel(:ncol,:), & !in
+       vmr(:,:,o3_ndx)) !in-outs
 
     !-----------------------------------------------------------------------      
     !         ... Check for negative values and reset to zero
     !-----------------------------------------------------------------------      
     call negtrc( vmr, ncol )
 
-    !-----------------------------------------------------------------------      
-    !         ... Set upper boundary mmr values
-    !-----------------------------------------------------------------------      
-    call set_fstrat_vals( vmr, pmid, pint, troplev, calday, ncol,lchnk )
-
-    !-----------------------------------------------------------------------      
-    !         ... Set fixed lower boundary mmr values
-    !-----------------------------------------------------------------------      
-    call flbc_set( vmr, ncol, lchnk, map2chm )
-
-    if ( ghg_chem ) then
-       call ghg_chem_set_flbc( vmr, ncol )
-    endif
-
     !-----------------------------------------------------------------------      
     !         ... Xform from vmr to mmr
     !-----------------------------------------------------------------------      
     call vmr2mmr( vmr, mmr_tend, mbar, ncol )
 
-    call set_short_lived_species( mmr_tend, lchnk, ncol, pbuf )
-
     !-----------------------------------------------------------------------      
     !         ... Form the tendencies
     !----------------------------------------------------------------------- 
-    do m = 1,gas_pcnst 
-       mmr_new(:ncol,:,m) = mmr_tend(:ncol,:,m)
-       mmr_tend(:ncol,:,m) = (mmr_tend(:ncol,:,m) - mmr(:ncol,:,m))*delt_inverse
+    do mm = 1,gas_pcnst 
+       mmr_new(:ncol,:,mm) = mmr_tend(:ncol,:,mm)
+       mmr_tend(:ncol,:,mm) = (mmr_tend(:ncol,:,mm) - mmr(:ncol,:,mm))*delt_inverse
     enddo
 
-    do m = 1,pcnst
-       n = map2chm(m)
-       if( n > 0 ) then
-          qtend(:ncol,:,m) = qtend(:ncol,:,m) + mmr_tend(:ncol,:,n) 
+    do mm = 1,pcnst
+       nn = map2chm(mm)
+       if( nn > 0 ) then
+          qtend(:ncol,:,mm) = qtend(:ncol,:,mm) + mmr_tend(:ncol,:,nn) 
```

Comment:
Very minor -- 'end if' and 'end do' here, and 'end do' line 585.

Reply:
I have fixed these and also cleaned up and unified some more codes.